scipy/stats/_continuous_distns.py

```diff
@@ -3313,6 +3314,312 @@ def _ppf(self, q, k, s):
+
+        def f2(x, h, k):
+            '''cdf = (1.0 - h*exp(-x))**(1.0/h)
+               logcdf = ...
+            '''
+            return (1.0/h)*special.log1p(-h*exp(-x))
+
+        def f3(x, h, k):
+            '''cdf = exp(-exp(-x))
+               logcdf = ...
+            '''
+            return -exp(-x)
+
+        return _lazyselect(condlist,
+                           [f0, f1, f2, f3],
+                           [x, h, k],
+                           default=nan)
+
+    def _ppf(self, q, h, k):
+        condlist = [np.logical_and(h != 0, k != 0),
+                    np.logical_and(h == 0, k != 0),
+                    np.logical_and(h != 0, k == 0),
+                    np.logical_and(h == 0, k == 0)]
+
+        def f0(q, h, k):
+            return 1.0/k*(1.0 - ((1.0 - (q**h))/h)**k)
+
+        def f1(q, h, k):
+            return 1.0/k*(1.0 - (-log(q))**k)
+
+        def f2(q, h, k):
+            '''ppf = -log((1.0 - (q**h))/h)
+            '''
+            return -special.log1p(-(q**h)) + log(h)
+
+        def f3(q, h, k):
+            return -log(-log(q))
+
+        return _lazyselect(condlist,
+                           [f0, f1, f2, f3],
+                           [q, h, k],
+                           default=nan)
+
+    def _stats(self, h, k):
+        if h >= 0 and k >= 0:
+            maxr = 5
+        elif h < 0 and k >= 0:
+            maxr = int(-1.0/h*k)
+        elif k < 0:
+            maxr = int(-1.0/k)
+        else:
+            maxr = 5
+
+        outputs = [None if r < maxr else nan for r in range(1, 5)]
+        return outputs[:]
+kappa4 = kappa4_gen(name='kappa4')
+
+
+class kappa3_gen(rv_continuous):
+    """Kappa 3 parameter distribution.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `kappa` is::
+
+        kappa3.pdf(x, a) =
+            a*[a + x**a]**(-(a + 1)/a),     for ``x > 0``
+            0.0,                            for ``x <= 0``
+
+    `kappa3` takes ``a`` as a shape parameter and ``a > 0``.
+
+    References
+    ----------
+    P.W. Mielke and E.S. Johnson, "Three-Parameter Kappa Distribution Maximum
+    Likelihood and Likelihood Ratio Tests", Methods in Weather Research,
+    701-707, (September, 1973),
+    http://docs.lib.noaa.gov/rescue/mwr/101/mwr-101-09-0701.pdf
+
+    B. Kumphon, "Maximum Entropy and Maximum Likelihood Estimation for the
+    Three-Parameter Kappa Distribution", Open Journal of Statistics, vol 2,
+    415-419 (2012)
+    http://file.scirp.org/pdf/OJS20120400011_95789012.pdf
+
+    %(after_notes)s
+
+    %(example)s
+
+    """
+    def _argcheck(self, a):
+        return a > 0
+
+    def _pdf(self, x, a):
+        x = asarray(x)
+        ret = np.zeros(shape(x), dtype=x.dtype)
+        ret[x > 0] = a*(a + x**a)**(-1.0/a-1)
+        return ret
+
+    def _cdf(self, x, a):
+        x = asarray(x)
+        ret = np.zeros(shape(x), dtype=x.dtype)
+        ret[x > 0] = x*(a + x**a)**(-1.0/a)
```

Comment:
Actually, no. The CI failure is (you need to click on `get the raw log` at the bottom of the page)
```
======================================================================
ERROR: test_continuous_basic.test_cont_basic(<scipy.stats._continuous_distns.kappa3_gen object at 0x7fcbb8d54278>, (1.0,), 'kappa3')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/scipy/stats/tests/common_tests.py", line 93, in check_entropy
    ent = distfn.entropy(*arg)
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/scipy/stats/_distn_infrastructure.py", line 1040, in entropy
    place(output, cond0, self.vecentropy(*goodargs) + log(scale))
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/numpy/lib/function_base.py", line 1874, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/numpy/lib/function_base.py", line 1940, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/scipy/stats/_distn_infrastructure.py", line 2232, in _entropy
    h = integrate.quad(integ, self.a, self.b)[0]
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/scipy/integrate/quadpack.py", line 315, in quad
    points)
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/scipy/integrate/quadpack.py", line 382, in _quad
    return _quadpack._qagie(func,bound,infbounds,args,full_output,epsabs,epsrel,limit)
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/scipy/stats/_distn_infrastructure.py", line 2227, in integ
    val = self._pdf(x, *args)
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/scipy/stats/_continuous_distns.py", line 3605, in _pdf
    ret[x > 0] = a*(a + x**a)**(-1.0/a-1)
IndexError: 0-d arrays can't be indexed.
---------------------------------------------------